### DISCUSSION TITLE
How to find a locality when it isn’t included in address details?

### DISCUSSION THREAD

**Discussion by daBrado** [osm-search/Nominatim#3080] [Q&A]
Sometimes I do a search that is definitely in a certain locality, but whose address details doesn’t include it. I was wondering if there is a way to deal with this situation.

For example, the query:

https://nominatim.openstreetmap.org/search.php?street=6242+Silverton+Way&state=California&format=jsonv2&addressdetails=1

…gets me the address I’m interested in, but doesn’t include its locality of Carmichael:

https://nominatim.openstreetmap.org/ui/details.html?osmtype=W&osmid=33234518&class=boundary

So, I guess I’m wondering, how can I get to Carmichael from the original search result? E.g. is there a way to find all those administrative boundaries that contain a point…?

**Comment by lonvia** [comment]
There are a couple of problems with the tagging of Carmichael:

First, it has a `boundary=administrative` tag but there is no `admin_level`, so Nominatim wouldn't know where to put it in the hierarchy of administrative boundaries. It will fall back to using the lowest level but it is still better to explicitly state the level. If Carmichael is not really an administrative entity but only an informal neighbourhood, you can also simply remove the boundary tag. Having a `place` tag is sufficient.

Second, `place=locality` is defined [in the wiki](https://wiki.openstreetmap.org/wiki/Tag:place%3Dlocality) as an unpopulated place. This is why Nominatim ignores them when creating an address. Carmichael is clearly inhabited. Change it to one of the [tags for inhabited urban places](https://wiki.openstreetmap.org/wiki/Key:place#Populated_settlements,_urban) like `place=neighborhood` or `place=quarter` and Nominatim will pick it up.

**Reply by daBrado** [reply]
Interesting, thanks for the information.

So, I guess one should edit that entry / those entries on OpenStreetMap.org ... I'll try looking at that, though honestly I'm getting a little confused about what exactly it should be changed to... Wikipedia says it is a "census-designated place" and I'm not sure how that translates to OSM terminology. (Not sure that it is actually a "neighborhood" or a "quarter" because it doesn't seem to be in another city, though it is closely aligned with the city of Sacramento...)

Though, I'm also still curious, for these situations where perhaps something is mislabeled but there still is some information to be gleaned from the various areas that enclose the point of interest; is it possible to query for a list of those areas for a given point?

**Reply by lonvia** [reply]
If  you are unsure about the tagging, I suggest you discuss with the US OSM community first. [Slack](https://osmus.slack.com/) is likely the best place but [forum](https://community.openstreetmap.org/c/communities/us/78) and [mailinglist](https://lists.openstreetmap.org/listinfo/talk-us/) work, too.

As for finding problematic areas. There is a [QA view](https://nominatim.org/qa/#map=5.47/38.40/-101.39&layer=no_admin_level) for the first problem I mentioned, missing admin_level. It is more difficult to find inhabited places tagged as place=locality because there is no clear definition of "inhabited". You can use the Overpass API to find objects if you have a good definition. For example, to find all the objects that have the suspicious combination of `boundary=administrative` and `place=locality` around Sacramento run this: https://overpass-turbo.eu/s/1weA

**Reply by daBrado** [reply]
Thanks for the pointers!

As for problematic areas, I think maybe we're talking about different things; I wasn't talking about identifying problematic areas, but instead, in the case I come across such areas, still being able to extract additional not-so-categorized location information based on overlapping boundaries that isn't necessarily tagged in a way Nominatim likes.  But, perhaps I'm asking about something that doesn't make much sense w.r.t. how Nominatim operates.
